src/Core/Utilities/CoreHelpers.cs

```diff
@@ -798,16 +798,21 @@ public static string ObfuscateEmail(string email)
             return email;
         }
 
-        var sb = new StringBuilder();
-        sb.Append(emailParts[0][..2]);
-        for (var i = 2; i < emailParts[0].Length; i++)
+        var finalLength = emailParts[0].Length + 1 + emailParts[1].Length;
```

Comment:
If you're up for it I think we can get rid of even more allocations. We could switch the `email.Split('@', ...)` to use a little helper I added here: https://github.com/bitwarden/server/blob/363dd6493a3996e1b9ef8522504259e60f4d7f31/src/Core/Utilities/SpanExtensions.cs#L5-L20

If you're up for that let me know but if not you can just sign the CLA and I can work on getting this merged.

Reply:
Sure! Let me take another attempt. I wasn't sure whether the `'Split(char, StringSplitOptions.RemoveEmptyEntries)` was there to handle any edge cases but thinking through it a bit more I should be able to make some more improvements.